log4j2-propagation/src/main/java/nl/talsmasoftware/context/log4j2/ThreadContextData.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016-2021 Talsma ICT
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.talsmasoftware.context.log4j2;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.ThreadContext;
+
+/**
+ * Represents a snapshot of the data of the Log4j 2 {@link ThreadContext} of a
+ * specific thread at a certain point in the past.
+ */
+public class ThreadContextData {
+    private final Map<String, String> contextMap;
+    private final List<String> contextStack;
+
+    /**
+     * Applies the data to the current thread.
+     *
+     * @param data data to apply, may be {@code null}
+     * @param overwrite whether all existing data should overwritten
+     */
+    static void applyToCurrentThread(ThreadContextData data, boolean overwrite) {
```

Comment:
There is an instance method `applyToCurrentThread()` as well, which could be `private`, but is package-private for testing (maybe I should rewrite the tests to not use that method though).
The reason why this method exists is similar to what I described in https://github.com/talsma-ict/context-propagation/pull/207#discussion_r577189357, I wanted to have logic related to manipulation of Log4j 2's `ThreadContext` only in `ThreadContextData` because I feel it belongs more there than to `ThreadContextManager`. And because the `data` can be `null`, only having an instance method would not work.